### ISSUE TITLE
Why the RAM used keep growing in the training?

### ISSUE THREAD

**Issue by newoneincntk** [Microsoft/CNTK#453]
I try to train a regression network by using  SimpleNetworkBuilder and CNTKTextFormatReader. I found that when a epoch finished, the RAM used would not be  released, and when a new epoch started the whole training data would be read into ram again, so the RAM used keeped growing until there was no free RAM. 
The cntk config file is
# Parameters can be overwritten on the command line
# for example: cntk configFile=myConfigFile RootDir=../.. 
# For running from Visual Studio add
# currentDirectory=$(SolutionDir)/<path to corresponding data folder> 
RootDir = ".."

ConfigDir = "$RootDir$/Config"
DataDir = "$RootDir$/Data"
OutputDir = "$RootDir$"
ModelDir = "$RootDir$/Output/Models"

# deviceId=-1 for CPU, >=0 for GPU devices, "auto" chooses the best GPU, or CPU if no usable GPU is available
deviceId = 0

command = Simple_Demo_Train:Simple_Demo_Test:Simple_Demo_Output

precision = "float"
traceLevel = 1
modelPath = "$ModelDir$/simple.dnn"
#outputNodeNames = Hlast

#######################################
#  TRAINING CONFIG                    #
#######################################

Simple_Demo_Train = [
    action = "train"

    # Notation xxx:yyy*n:zzz is equivalent to xxx, then yyy repeated n times, then zzz
    # Example: 10:20*3:5 is equivalent to 10:20:20:20:5
    SimpleNetworkBuilder = [
        # 2 input, 2 50-element hidden, 2 output
        layerSizes = 129:512*3:129
        trainingCriterion = "SquareError"
      #  trainingCriterion="CrossEntropyWithSoftmax"
        evalCriterion = "ErrorPrediction"
       # layerTypes = "Sigmoid"
       layerTypes="RectifiedLinear"
        initValueScale = 1.0
        applyMeanVarNorm = true
        uniformInit = true
        needPrior = true
    ]

    SGD = [
	    # epochSize = 0 means epochSize is the size of the training set 
        epochSize = 0 
        minibatchSize = 128
        learningRatesPerMB = 0.1*10:0.01*10:0.001
        momentumPerMB = 0.9
        dropoutRate = 0.0
        maxEpochs = 30
    ]
       
    # Parameter values for the reader
    reader = [
        readerType = "CNTKTextFormatReader"
        file = "$DataDir$/SimpleDataTrain.txt"
        miniBatchMode = "partial"
        randomize = "auto"
        #chunkSizeInBytes = 33554432
        #numChunksToCache = 16
        # verbosity = 1  
        traceLevel = 1
       input=[
        features = [
            alias="f"
            dim = 129       # two-dimensional input data
            format="dense"
        ]

        labels = [
            alias="l"
            dim = 129        # One label dimension
            format="dense"
      
        ]
      ]
    ]
]

########################################
#  TEST RESULTS                        #
#  (computes prediction error and      #
#   perplexity on a test set and       #
#   writes the output to the console.) #
########################################

Simple_Demo_Test = [
    action = "test"

    # Parameter values for the reader
    reader = [
        readerType = "CNTKTextFormatReader"
        file = "$DataDir$/SimpleDataTest.txt"
        miniBatchMode = "partial"
        randomize = "auto"
        # verbosity = 1  
        traceLevel = 1
        input=[
        features = [
            alias="f"
            dim = 129       # two-dimensional input data
            format="dense"
        ]

        labels = [
            alias="l"
            dim = 129        # One label dimension
            format="dense"
      
        ]
      ]
    ]
]

########################################
#  OUTPUT RESULTS                      #
#  (Computes the labels for a test set #
#   and writes the results to a file.) #
########################################

Simple_Demo_Output=[
    action = "write"

    # Parameter values for the reader
    reader = [
        readerType = "CNTKTextFormatReader"
        file = "$DataDir$/SimpleDataTest.txt"
        miniBatchMode = "partial"
        randomize = "auto"
        # verbosity = 1  
        traceLevel = 1
        input=[
        features = [
            alias="f"
            dim = 129       # two-dimensional input data
            format="dense"
        ]

        labels = [
            alias="l"
            dim = 129        # One label dimension
            format="dense"
      
        ]
      ]
    ]
   outputNodeNames = Hlast
	
    outputPath = "$OutputDir$/SimpleOutput"    # Dump output as text
]

Simple_Demo_Dumpnode = [
    action = "dumpnode"
outputPath = "$OutputDir$/node"    # Dump output as text
]

**Comment by raaaar** [comment]
Are you running the toolkit built with the latest changes from master? If not, could you please pull and try again? This issues was addressed a few days ago by 52166742d90ebe6da4d0cf543d9aab22c552afd0.

**Comment by newoneincntk** [comment]
Thank you for your advise, the above-mentioned bug is in the CNTK1.1 binary release, now I ran the latest package and it works well.
